jlm/hls/backend/rvsdg2rhls/add-forks.cpp

```diff
@@ -36,14 +37,15 @@ add_forks(jlm::rvsdg::region * region)
         add_forks(structnode->subregion(n));
       }
     }
+    bool isConstant = rvsdg::is<rvsdg::bitconstant_op>(node->operation())
+                   || rvsdg::is<rvsdg::ctlconstant_op>(node->operation());
```

Comment:
This will only work for those constants, but what about floating point constants or other constants?

I guess a better way is to check whether a node has no input to determine whether it is a constant.

Reply:
Changed to checking if ninputs() == 0